input/pagecontent/algorithms.md

```diff
@@ -1,257 +1,189 @@
 
-PV1\|1\|O\|\|
+**Generalize categories with small counts:** Small counts of data subjects for a specific value of a numeric quasi-identifier increase the risk of re-identification. To mitigate this risk, categories with low counts can be generalized into a single category with a higher count. For example, if there is only one patient aged 80, 85, and 90, respectively, these individual age categories can be combined into a broader "80+" category. This technique could also be applied in contingency tables with several categories have small count values. For example, For example, rather than re-porting that there are two people with blue eyes, one person with green eyes, and one person with hazel eyes, it may be reported that there are four people with blue, green,or hazel eyes[(NIST_SP_800-188_2023)](references.html#NIST_SP_800-188_2023).
 
-IN1\|\|Plan123\|PART\|InsureCo\|Address1\|Admin\|+1-801-555-1212\|Group12\|GroupNm\|EmpID\|CoNm\|20080101\|20081231\|Auth\|TypeP\|Spencer^Royce\|Son\|19990101\|Addr\|AOB\|COB\|\|\|\|\|\|\|\|\|\|\|\|\|\|\|""
+**Noise addition:** Also called "noise infusion," and "noise injection" (and occasion-ally "partially synthetic data"), this approach adds small random values to attributes. For example, instead of reporting that a person is 84 years old, the person may be reported as being 79 years old[(NIST_SP_800-188_2023)](references.html#NIST_SP_800-188_2023).
 
-```
+##### De-Identifying Dates
 
-**A08 after replacing values with fixed values:**
+Dates are a common quasi-identifier and must be handled carefully to prevent re-identification. Common techniques include:
 
-```hl7
-MSH\|^\~\\&\|PROACCESS5\|DHIN\|BIOSENSE\|CDC01\|20080808290000\|\|ADT^A08\|111122223333\|P\|2.5\|\|
+* **Date Shifting:** All dates for a specific individual are shifted by a random (but consistent) number of days. This preserves the interval between events but obscures the actual dates.
+* **Coarsening:** The precision of the date is reduced. For example, a specific birth date (e.g., "1985-04-12") is replaced with just the year ("1985") or a broader range (e.g., "1985-1990").
+* **Converting to Age:** Dates can be converted into an age at the time of an event. This age can then be generalized (e.g., grouped into 5-year brackets) to further reduce risk.
 
```

Comment:
previously we also recognized moving to week-of-year

You don't mention time. But there are times that need to be preserved while others can be removed. Previously we recognized that some times relative to the meals, sleep, or activities.

Reply:
an attack that we do need to address is where an attacker knows when a subject has visited various medical appointments, from that they can look at de-identified data and correlate which pseudonym is their subject.

Times would be best to remove, and most of the time they are not important to the use-case for the de-identified dataset. But sometimes times are important. It is at these cases that one should determine the method they will use. Generalization (moving times to some fixed but useful times of the day), fuzzing (random adjustment that is not bigger than the significant purpose of the time, etc.